### ISSUE TITLE
ARM64 ssl module in Python is not available (from PIP)

### ISSUE THREAD

**Issue by peardox** [bytedeco/javacpp-presets#796]
This happens on my Nano )prob affects other platforms as well) when in gets to Numpy. Seen this before so had a clue as to the answer.

The previous step in to build CPython which will happily build with or without SSL/TLS but if you build without SSL/TLS the Numpy fails...

On Ubuntu you get the deps with...
sudo apt-get install libreadline-gplv2-dev libncursesw5-dev libssl-dev libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev

Dunno if they're all required (StackOverflow answer)

Prob a good idea to stick a check in for SSL dev libs while building CPython

**Comment by saudet** [comment]
I'm pretty sure it gets enabled properly, for example, see this build log:
> checking for aarch64-linux-gnu-pkg-config... /usr/bin/aarch64-linux-gnu-pkg-config
> checking whether compiling and linking against OpenSSL works... yes
> checking for X509_VERIFY_PARAM_set1_host in libssl... yes
> checking for --with-ssl-default-suites... python

https://api.travis-ci.org/v3/job/592363041/log.txt

What error messages are you getting?

**Comment by peardox** [comment]
I'll reverse the solution when I get a chance to repeat the error + send you the output

Kindly bear with me - I wanna do a clean build later / tomorrow

**Comment by peardox** [comment]
OK - one extremely long build log attached if you wanna examine it

I removed libssl-dev before starting this run

The highlight is @ line 31391... Part of the CPython build

Could not build the ssl module!
Python requires an OpenSSL 1.0.2 or 1.1 compatible libssl with X509_VERIFY_PARAM_set1_host().
LibreSSL 2.6.4 and earlier do not provide the necessary APIs, https://github.com/libressl-portable/portable/issues/381

A lot above this @ 13665 it actually build a static libssl as bart of FFMpeg :)

Things then continue to the 45000 area where Python gets upset that it ain't got SSL

From here on in adding libssl-dev back in and restarting :cpython runs thru this lot again until #797 where tesseract gets upset at -ljbig

[build.01.zip](https://github.com/bytedeco/javacpp-presets/files/3692845/build.01.zip)

**Comment by saudet** [comment]
It won't build without libssl-dev, that's expected.

**Comment by saudet** [comment]
If I understand correctly, this is only a documentation issue, right?
We would need to update the wiki page about that actually:
https://github.com/bytedeco/javacpp-presets/wiki/Build-Environments
Based on the scripts here doing cross compilation:
https://github.com/bytedeco/javacpp-presets/blob/master/ci/install-arm.sh
https://github.com/bytedeco/javacpp-presets/blob/master/ci/install-ppc.sh

**Comment by peardox** [comment]
The point I'm making about missing libssl-dev is that the cause of the error isn't obvious - far from it.

The error is reported when building Numpy but the cause of the error is the CPython build missing libssl-dev

I've got Python + PIP installed and they've got SSL support, this is what threw me at first as it's not using the system Python / PIP but one it installs itself which will not have SSL support without libssl-dev. When building CPython a check should be made for libssl-dev and die if missing.

I never do / have never done cross compilation - that's why I suggested getting someone to let the project have a PPC cloud box (someone says yes eventually). Same goes for Power8/9 and upcoming ARM servers.

I'll get around to looking into Travis-CI sometime soon
